### ISSUE TITLE
I don't see a caller number on Android

### ISSUE THREAD

**Issue by zishe** [priteshrnandgaonkar/react-native-call-detection#30]
I created an app with this package and it works on RN 0.58 https://github.com/zishe/call-detector . The only thing is that I don't see a caller number. Code is almost the same as in your example.

```
new event:  Incoming  -  
new event:  Missed  -  null
```

**Comment by DragonRaider5** [comment]
Is this on Android or iOS?

Alexander Kireev <notifications@github.com> schrieb am Mi., 27. Feb. 2019,
15:45:

**Comment by zishe** [comment]
Android

**Comment by zishe** [comment]
Maybe because it was Java 11. I'll check with 8th version.

**Comment by DragonRaider5** [comment]
Thanks for your testing, please hit me back! :+1:

Alexander Kireev <notifications@github.com> schrieb am Mi., 27. Feb. 2019,
17:59:

**Comment by zishe** [comment]
@DragonRaider5 still the same. Is it because of a new version?

**Comment by zishe** [comment]
I just checked you example - same result, it doesn't show a number from which I call.

**Comment by zishe** [comment]
Finally, you need to have `READ_CALL_LOG` permissions. I wrote next function for requesting them:

```
async function requestPermission() {
  try {
    const granted = await PermissionsAndroid.request(
      PermissionsAndroid.PERMISSIONS.READ_PHONE_STATE,
      {
        title: 'Need App Permission',
        message: 'App needs access to your phone state',
        buttonNeutral: 'Ask Me Later',
        buttonNegative: 'Cancel',
        buttonPositive: 'OK',
      },
    );
    const logGranted = await PermissionsAndroid.request(
      PermissionsAndroid.PERMISSIONS.READ_CALL_LOG,
      {
        title: 'Need App Permission',
        message: 'App needs access to your call log',
        buttonNeutral: 'Ask Me Later',
        buttonNegative: 'Cancel',
        buttonPositive: 'OK',
      },
    );

    if (granted === PermissionsAndroid.RESULTS.GRANTED && logGranted === PermissionsAndroid.RESULTS.GRANTED) {
      console.log('Permissions granted');
      return true;
    } else {
      console.log('Permissions denied');
      return false;
    }
  } catch (err) {
    console.warn(err);
    return false;
  }
}
```

And I've added it to manifest file:

```
    <uses-permission android:name="android.permission.READ_CALL_LOG"/>
```

**Comment by zishe** [comment]
You can check my example project https://github.com/zishe/call-detector and it would be great if you actualize all versions. I can't help you at the moment, cause of too much work.

**Comment by zishe** [comment]
Ugh, hack, but I need to find out the phone number by which I call. It's null at the moment.

**Comment by zishe** [comment]
> Ugh, hack, but I need to find out the phone number by which I call. It's null at the moment.

@DragonRaider5 does it work for you?

**Comment by zishe** [comment]
@DragonRaider5 https://github.com/priteshrnandgaonkar/react-native-call-detection/blob/aacf9229071c12a50317dacf9ea5006af4d4e354/android/src/main/java/com/pritesh/calldetection/CallDetectionManagerModule.java#L130 why it's null here?

**Comment by zishe** [comment]
Oh, and on my devise it doesn't show phone number too - Android 8.1

**Comment by DragonRaider5** [comment]
Because phone number detection was only developed for incoming calls.

I'd not call updating permissions to match android changes a hack.
Currently I'm very busy too, but I'll try it as soon as possible.

In addition, we'd need @priteshrnandgaonkar to push the updates to npm.

**Comment by zishe** [comment]
@DragonRaider5 I understand. Well, everything works in my fork https://github.com/zishe/react-native-call-detection/ , if someone need this solutions he can get it from there.

I can close it if this is not an issue.

**Comment by priteshrnandgaonkar** [comment]
Sorry for the delayed response. @DragonRaider5, I can publish an new update to npm

**Comment by priteshrnandgaonkar** [comment]
@zishe, can you put up the PR to fix the issue.

**Comment by priteshrnandgaonkar** [comment]
@DragonRaider5, Published 1.7.1.

**Comment by guhyeon** [comment]
```
Would you pass phoneNumber as parameter in all state?

It will be easily possible just change one method. thank you.

file : android\src\main\java\com\priteh\calldetection\ CallDetectionManagerModule.Java

to be code:
@Override
    public void phoneCallStateUpdated(int state, String phoneNumber) {
        jsModule = this.reactContext.getJSModule(CallStateUpdateActionModule.class);

        switch (state) {
            //Hangup
            case TelephonyManager.CALL_STATE_IDLE:
                if(wasAppInOffHook == true) { // if there was an ongoing call and the call state switches to idle, the call must have gotten disconnected
                    jsModule.callStateUpdated("Disconnected", phoneNumber);
                } else if(wasAppInRinging == true) { // if the phone was ringing but there was no actual ongoing call, it must have gotten missed
                    jsModule.callStateUpdated("Missed", phoneNumber);
                }

                //reset device state
                wasAppInRinging = false;
                wasAppInOffHook = false;
                break;
            //Outgoing
            case TelephonyManager.CALL_STATE_OFFHOOK:
                //Device call state: Off-hook. At least one call exists that is dialing, active, or on hold, and no calls are ringing or waiting.
                wasAppInOffHook = true;
                jsModule.callStateUpdated("Offhook", phoneNumber);
                break;
            //Incoming
            case TelephonyManager.CALL_STATE_RINGING:
                // Device call state: Ringing. A new call arrived and is ringing or waiting. In the latter case, another call is already active.
                wasAppInRinging = true;
                jsModule.callStateUpdated("Incoming", phoneNumber);
                break;
        }
    }
```
